### DISCUSSION TITLE
SETTINGS subsystem(CONFIG_SETTINGS) with FCB (SETTINGS_FCB) storage back-end does not work on STM32U575

### DISCUSSION THREAD

**Discussion by zafersn** [zephyrproject-rtos/zephyr#75056] [Q&A]
tested setting subsys module (CONFIG_SETTINGS) on the stm32u575 and settings_backend_init returns an error `<err> flash_stm32: Write range invalid. Offset: 983040, len: 8` . so went into this problem and stm32u5 doesnt accept if written data' offeset and len is not a multiple of FLASH_STM32_WRITE_BLOCK_SIZE (16 bytes)

![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/fc1f0a1a-2e95-47c6-82d3-ad358cd00808)

and here is where the size is given. 

![image](https://github.com/zephyrproject-rtos/zephyr/assets/13236364/cd96edfd-4fe8-4795-8f1b-4ed38cd9ef7b)

sizeof(fda) = 8 bytes

So, am I wrong here, or is it an issue?

**Comment by Laczen** [comment]
@zafersn, this is a bug. FCB at the moment does not seem to support write-block-size > 8 as a result of how the sector header is written/read.

**Reply by zafersn** [reply]
@Laczen Thanks for your reply. Is there an issue created for this? I can follow it up  to stay updated? if you know, could you please mention here?

**Reply by Laczen** [reply]
I don't think this has been reported as an issue/bug. The fix should be quite trivial: create a buffer that is correctly sized `fcb_get_length(..., size_of(fda))`, copy the `fda` to the start of this buffer, and write the buffer.
